### PR TITLE
added /network/total-staked API route

### PR DESCRIPTION
**Pull request by miiu96**
Added `network/total-staked` API route
This API route will return total-staked value only if is called from a metachain node

### CODE REVIEW

**Review by sasurobert — CHANGES_REQUESTED** [review]

File: cmd/node/main.go [hunk 1]
```diff
@@ -16,6 +16,8 @@ import (
 	"syscall"
 	"time"
 
+	"github.com/ElrondNetwork/elrond-go/node/totalStakedAPI"
+
```
[the remark this answers is not in the document]
Comment (miiu96) [reply cmd/node/main.go]: done.

File: api/network/routes.go [hunk 2]
```diff
@@ -112,3 +116,39 @@ func EconomicsMetrics(c *gin.Context) {
 		},
 	)
 }
+
+// GetTotalStaked is the endpoint that will return the total staked value
+func GetTotalStaked(c *gin.Context) {
```
Comment (iulianpascalau) [inline api/network/routes.go:121]: unit test?

File: cmd/node/config/config.toml [hunk 3]
```diff
@@ -766,3 +766,6 @@
 
 [Logs]
     LogFileLifeSpanInSec = 86400
+
+[TotalStakedValueAPI]
+    TotalStakedValueAPICacheDurationInMinutes = 100
```
Comment (iulianpascalau) [inline cmd/node/config/config.toml:771]: No, the node should try not to cache values. That is the job of the middle layer that will query the node.

**Review by sasurobert — DISMISSED** [review]

File: node/totalStakedAPI/interface.go [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package totalStakedAPI
+
+import "math/big"
+
+// TotalStakedValueHandler defines the behavior of a component able to return total staked value
+type TotalStakedValueHandler interface {
```
Comment (iulianpascalau) [inline node/totalStakedAPI/interface.go:6]: Interfaces should be placed where are used, not where are implemented. Plese move this in node/external package

File: node/totalStakedAPI/totalStakedValueProcessor.go [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+	mutex            sync.RWMutex
+}
+
+// NewTotalStakedValueProcessor will create a new instance of totalStakedValueProcessor
+func NewTotalStakedValueProcessor(
+	marshalizer marshal.Marshalizer,
+	cacheDuration time.Duration,
+	accounts state.AccountsAdapter,
+) (*totalStakedValueProcessor, error) {
+	if cacheDuration <= 0 {
```
Comment (iulianpascalau) [inline node/totalStakedAPI/totalStakedValueProcessor.go:33]: as stated, no need for this cache duration

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

File: api/network/routes.go (hunk 2 above)
Comment (miiu96) [reply api/network/routes.go:121]: done.

File: node/totalStakedAPI/interface.go (hunk 4 above)
Comment (miiu96) [reply node/totalStakedAPI/interface.go:6]: done.

**Review by iulianpascalau — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by stefanandy — APPROVED** [review]
System test passed

**Pull request merged**